{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<b>fastANI</b> library counts ANI metric without align phase\n",
    "\n",
    "https://github.com/ParBLiSS/FastANI\n",
    "\n",
    "NOTE: No ANI output is reported for a genome pair if ANI value is much below 80%."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import os\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "from Bio import SeqIO\n",
    "from Bio.SeqRecord import SeqRecord"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Sort aligned 92 genes to 12 organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "path = '../data/fastANI_data/pre_aligned_genes_by_orgs/'\n",
    "gene_names = [f.split('.')[0] for f in os.listdir(path)]\n",
    "full_paths = [path + f for f in os.listdir(path)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['alaS',\n",
       " 'argS',\n",
       " 'aspS',\n",
       " 'cgtA',\n",
       " 'coaE',\n",
       " 'cysS',\n",
       " 'dnaA',\n",
       " 'dnaG',\n",
       " 'dnaX',\n",
       " 'engA']"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_names[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['fastANI_data/pre_aligned_genes_by_orgs/alaS.zZ.fasta',\n",
       " 'fastANI_data/pre_aligned_genes_by_orgs/argS.zZ.fasta',\n",
       " 'fastANI_data/pre_aligned_genes_by_orgs/aspS.zZ.fasta',\n",
       " 'fastANI_data/pre_aligned_genes_by_orgs/cgtA.zZ.fasta',\n",
       " 'fastANI_data/pre_aligned_genes_by_orgs/coaE.zZ.fasta']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_paths[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_ids = []\n",
    "orgs = {}\n",
    "\n",
    "for gene_file, gene_name in zip(full_paths, gene_names):\n",
    "    genes = SeqIO.parse(gene_file, 'fasta')\n",
    "    for gene in genes:\n",
    "        org_ids.append(gene.id)\n",
    "        if orgs.get(gene.id) is None:\n",
    "            orgs[gene.id] = []\n",
    "        orgs[gene.id].append((gene_name, gene.seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "org_ids = list(set(org_ids))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "path_orgs = '../data/fastANI_data/orgs_gene_aligned_1066/'\n",
    "\n",
    "for org_id in org_ids:\n",
    "    write_path = path_orgs + org_id + '.fasta'\n",
    "    records = []\n",
    "    \n",
    "    for (gene_id, gene_seq) in orgs[org_id]:\n",
    "        records.append(SeqRecord(gene_seq, id=gene_id, description='| '+org_id))\n",
    "    \n",
    "    with open(write_path, 'w') as output_file:\n",
    "        SeqIO.write(records, output_file, 'fasta')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "full_paths_orgs = [path_orgs + f for f in os.listdir(path_orgs) if f.endswith('.fasta')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta',\n",
       " 'fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta']"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_paths_orgs"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Count ANI metric with fastANI module"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0158777 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.296648 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102814 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0144978 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265508 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5911e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0204607 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.302671 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00011304 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192671 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.28332 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103871 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0210566 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277477 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000146027 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0169869 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.273324 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8976e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0222606 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.292955 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000158922 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0199703 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.283139 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8627e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0205364 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.268414 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100401 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0290319 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.210947 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.9952e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.021489 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.299602 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104548 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0198938 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26869 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7046e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191108 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258559 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.3438e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0194168 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.271192 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102444 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0214348 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.264568 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2531e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0183863 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.300925 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000155295 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0152821 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257915 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108496 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188064 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.301836 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103109 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202436 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.250644 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.1988e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0213066 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.284925 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100961 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0198029 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258418 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103368 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0167749 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.215082 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.676e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0140728 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277582 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010574 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_2_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0181312 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.2814 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00021057 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0205395 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.281717 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000110678 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0146643 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277215 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2943e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0223619 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.291484 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000107734 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195274 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.254499 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00014586 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0166382 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.260646 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7075e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0233805 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.304802 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000279331 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191627 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.292874 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000136874 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190205 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.260977 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000126593 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0189444 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258988 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101379 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0141342 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.237034 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 7.7977e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0232675 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259127 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9048e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_3_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0225267 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.240821 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103273 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0303822 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258781 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104335 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0142668 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259801 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000135815 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202753 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.236788 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.0273e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0187026 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.253275 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8731e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0207723 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.273549 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101619 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0178177 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255036 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9811e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0148322 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.241381 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2404e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0194876 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.260604 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000189513 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0397774 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.365079 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5745e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0181531 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.230938 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9218e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0157252 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262979 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7969e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_4_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.02263 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259079 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000214847 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196023 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.27097 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000120257 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0189875 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.287274 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000153748 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0153254 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.244485 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4107e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0187654 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262347 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00013903 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0182038 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.279687 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5771e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0171756 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.311247 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100929 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0167955 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.248669 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.3758e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0243687 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.326473 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104945 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.029032 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.293601 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000122427 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0178423 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.22011 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000186958 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0262061 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.275294 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103261 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_5_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0384476 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.269058 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108485 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196991 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.28379 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102186 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202406 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.294091 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104051 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0194491 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262762 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7524e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0161976 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.251611 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000138978 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0153034 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257706 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7374e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180853 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.301967 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101221 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0163576 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.269798 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8777e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0289176 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.291816 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102213 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0174276 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.331185 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104735 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0172109 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.240128 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000137398 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0199205 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.313439 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000209708 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_6_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195157 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.305464 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000107873 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200039 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.316084 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000249446 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0179483 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.242363 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104861 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185297 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.307125 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000131032 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193183 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.23414 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100943 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0224306 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.251742 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5076e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019244 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26429 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000117867 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0219146 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.301108 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010944 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0283739 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257891 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000105781 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196755 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.272775 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8221e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0144358 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.200532 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.0313e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196647 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257576 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.985e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_7_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185156 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.243116 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103325 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.018606 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.250948 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.3e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0203293 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267541 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102561 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0203721 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.242254 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2898e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180153 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.250059 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102661 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.023956 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.283207 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5922e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0161108 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258294 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100587 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200963 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.247022 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102035 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200062 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262519 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9205e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202783 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.264996 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010121 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0159946 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.224049 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9212e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190777 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.256221 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103588 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_8_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0198423 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.256661 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.3664e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192165 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267631 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000114272 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195935 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.261004 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9494e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0213886 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.316132 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8975e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.018432 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.24771 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103058 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196918 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.254582 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9805e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188544 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277842 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9745e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0194945 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.261688 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100903 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0208303 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.27325 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8446e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188189 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.274701 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.416e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.011269 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.224063 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.7125e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192888 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.275173 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000112625 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_9_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0273248 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259687 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000105093 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191022 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.218272 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.4926e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191579 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.220847 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.279e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0216917 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.205894 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.4263e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202903 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.210995 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6022e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197918 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.214843 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6685e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0142867 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.215751 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2773e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195989 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.202578 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2801e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0233871 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.242722 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108139 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0201262 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.214213 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6287e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0160907 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.187837 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2901e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188617 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.224837 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6668e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_10_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197607 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.21418 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.7594e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019093 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265164 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108659 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0179784 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.2476 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000132966 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0140343 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258856 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6266e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019995 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.240858 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.443e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0206499 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262722 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6783e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193832 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26838 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101572 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185222 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.252158 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000126938 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0198094 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.252518 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100384 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0702771 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.364095 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9539e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0122877 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.292358 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000156959 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0173944 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.289221 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000131569 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_11_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.033194 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255021 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.383e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200371 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.261793 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5434e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196522 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262438 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108947 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193121 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.244293 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000133415 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0137635 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.254325 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104204 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0155398 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257324 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104094 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.026835 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.27148 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8685e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.018815 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.243623 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.1278e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0207384 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.256937 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000199101 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0142007 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.264779 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100749 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.012031 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.218993 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9293e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0204252 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259497 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101868 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI/output_12_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193067 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.276888 sec\r\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000118145 sec\r\n"
     ]
    }
   ],
   "source": [
    "path_output = '../data/fastANI_data/orgs_gene_aligned_1066_fastANI/'\n",
    "\n",
    "frag_len = 100\n",
    "kmers = 8\n",
    "min_frag = 5\n",
    "\n",
    "for (p1, path_org1) in enumerate(full_paths_orgs):\n",
    "    for (p2, path_org2) in enumerate(full_paths_orgs):\n",
    "        output_file = path_output + 'output_' + str(p1+1) + '_' + str(p2+1) + '.txt'\n",
    "        !fastANI -q {path_org1} -r {path_org2} -o {output_file} --fragLen {frag_len} -k {kmers} --minFrag {min_frag}"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Optionally, users can also get a second .matrix file with identity values arranged in a phylip-formatted lower triangular matrix by supplying --matrix parameter. NOTE: No ANI output is reported for a genome pair if ANI value is much below 80%. Such case should be computed at amino acid level."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0181145 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.295849 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103283 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197337 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255264 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4834e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0138454 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.286735 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000112805 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200606 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255543 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9693e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0186118 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.27119 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6802e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0303572 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.34126 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8611e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0175144 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.295599 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000123306 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0229848 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.28197 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8669e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0186347 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277901 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9941e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0161336 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.224664 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.3559e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0210197 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.288057 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000105743 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_1_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192731 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262406 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4521e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188485 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257726 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2698e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0203252 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265148 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000126269 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0164885 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.251163 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000124328 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202868 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.250994 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000109965 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0179068 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.273534 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000127082 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0207883 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.256613 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100751 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0147817 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.24506 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.1633e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0141472 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.272905 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7827e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0150108 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.270922 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8426e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0141865 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.263579 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000170606 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196757 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262129 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000109264 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_2_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0167687 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262667 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108401 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0239301 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.325247 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000107073 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190733 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.282792 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000111712 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0207949 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.296236 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102692 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0162312 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.248492 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6892e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0182266 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.248508 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5311e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190256 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.283458 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104272 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0198359 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.296931 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000171193 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.022936 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.245046 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.1493e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.021551 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262999 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8877e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0176318 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.209308 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.0003e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0199862 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.266731 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000106219 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_3_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019009 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.253782 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103782 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0196088 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.25553 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103114 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.015025 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267291 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102679 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0390496 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.328664 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.9805e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0233977 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.264465 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000106292 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0227577 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.275149 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00011846 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0179324 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.254536 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108048 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0209029 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265891 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.654e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0230434 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.269072 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000109909 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0171595 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258021 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6435e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0177733 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.212704 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100649 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0158676 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262243 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.937e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_4_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0199356 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.260296 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000115581 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0189468 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.268973 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6589e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200858 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262493 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000113762 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0184957 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.245552 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6214e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0176965 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.278768 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00011085 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193956 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259007 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8526e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0194315 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.268096 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7447e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197009 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.243516 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.3981e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0219371 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.272062 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000128496 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0141433 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262005 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8868e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0155895 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.214098 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6481e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191845 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259909 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000106753 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_5_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0237808 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257679 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000124361 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019231 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.279576 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100101 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190259 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.27152 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103349 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193609 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265149 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000106452 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0227036 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258511 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.543e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193981 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.271139 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8556e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191551 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.281277 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100082 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0200906 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.262187 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7083e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193901 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26539 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100726 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0172164 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.286804 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104263 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0120787 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.224264 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6346e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180272 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.281975 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000113025 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_6_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0214747 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267621 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104871 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0203065 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.313753 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000238807 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0275086 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259578 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.2144e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.017447 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.290314 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000114468 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202995 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.2509 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000137441 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193404 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.246661 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.0938e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195342 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.269218 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104574 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.033355 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.307259 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103777 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180597 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258148 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000143603 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0213832 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267245 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5674e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0162493 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.214453 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 7.954e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0175197 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.267835 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000225752 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_7_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0171361 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.248271 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100566 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019363 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.250973 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4499e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0195333 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26479 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100482 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0206965 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.244814 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000109906 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.022759 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255426 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103454 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0255545 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265155 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4846e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180887 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.277817 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100818 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0165874 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.241316 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100252 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.018073 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.273359 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101032 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0370699 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.333253 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000104459 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0184141 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.222501 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000108618 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0171573 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.261768 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000106261 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_8_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.022483 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.25659 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5162e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185036 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.261835 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000128801 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185416 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.270787 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8405e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192924 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258447 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010073 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0136511 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.264049 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9826e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0141529 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265502 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8848e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0145591 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.274775 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9699e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0192337 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.283791 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000102525 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0193651 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.254079 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.898e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0202015 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.275668 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.5353e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0224734 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.210967 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.9987e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0190568 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.280886 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000114484 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_9_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.030509 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.258536 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010307 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0199958 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.216675 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.222e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197403 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.22918 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.0001207 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0183578 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.20366 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2989e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0140379 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.216393 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.5735e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0167479 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.217525 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.5985e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0188515 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.218332 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.6579e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0158108 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.225013 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2269e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0183598 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257234 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7383e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0208509 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.231034 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.906e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0288334 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.219047 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.2713e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0191791 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.220436 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.8902e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_10_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0179303 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.21037 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.7095e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0185694 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.275963 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6284e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019323 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.249259 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.9262e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.014286 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.248204 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8132e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0272097 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.24273 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000112712 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0183676 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259789 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.7067e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0149279 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.273781 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8925e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0189866 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.270666 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000234632 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0357395 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.322333 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000134988 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0338634 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.384295 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000128307 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0217579 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259995 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 8.3828e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0211737 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.285745 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000103219 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_11_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0197682 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.255351 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.1612e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_sonorensis_GCF_002202015.1_ASM220201v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_1.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56888\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19781\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7203) ... (42, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0256363 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.265941 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.4733e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_halotolerans_GCF_001517105.1_ASM151710v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_2.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56895\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20327\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7391) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.019154 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.296663 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000109989 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_licheniformis_GCF_000011645.1_ASM1164v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_3.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56952\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19905\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7269) ... (41, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0176264 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.280332 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.00010617 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_mojavensis_GCF_000245335.1_ASM24533v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_4.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 54015\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20025\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7598) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0173973 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.25525 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000127571 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_tequilensis_GCF_000507145.1_KCTC_13622_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_5.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56876\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20306\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7375) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0166048 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.292357 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000146278 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_amyloliquefaciens_GCF_000196735.1_ASM19673v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_6.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56944\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19997\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7180) ... (39, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0524721 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.307967 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000120239 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_paralicheniformis_GCF_000408885.1_ASM40888v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_7.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56994\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 19866\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7196) ... (35, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0180215 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.259529 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000182327 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_subtilis_GCF_000009045.1_ASM904v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_8.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56874\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20334\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7402) ... (34, 2)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0231032 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.26338 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000119826 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_velezensis_GCF_002117165.1_ASM211716v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_9.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56981\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20006\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7210) ... (38, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0208062 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.257418 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.8851e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_vallismortis_GCF_000245315.1_ASM24531v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_10.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 43233\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 18130\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7855) ... (29, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0149746 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.225261 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 9.6066e-05 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_siamensis_GCF_000262045.1_KCTC_13613_01_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_11.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 57013\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20007\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7206) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.018725 sec\n",
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.284058 sec\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000101562 sec\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "Reference = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Query = [fastANI_data/orgs_gene_aligned_1066/Bacillus_atrophaeus_GCF_000742675.1_ASM74267v1_genomic.fasta]\n",
      "Kmer size = 8\n",
      "Fragment length = 100\n",
      "Threads = 1\n",
      "ANI output file = fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/output_12_12.txt\n",
      ">>>>>>>>>>>>>>>>>>\n",
      "INFO [thread 0], skch::Sketch::build, minimizers picked from reference = 56936\n",
      "INFO [thread 0], skch::Sketch::index, unique minimizers = 20243\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, Frequency histogram of minimizers = (1, 7226) ... (37, 1)\n",
      "INFO [thread 0], skch::Sketch::computeFreqHist, With threshold 0.001%, consider all minimizers during lookup.\n",
      "INFO [thread 0], skch::main, Time spent sketching the reference : 0.0204216 sec\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO [thread 0], skch::main, Time spent mapping fragments in query #1 : 0.266102 sec\r\n",
      "INFO [thread 0], skch::main, Time spent post mapping : 0.000100549 sec\r\n"
     ]
    }
   ],
   "source": [
    "path_output_m = '../data/fastANI_data/orgs_gene_aligned_1066_fastANI_matrix/'\n",
    "\n",
    "frag_len = 100\n",
    "kmers = 8\n",
    "min_frag = 5\n",
    "\n",
    "for (p1, path_org1) in enumerate(full_paths_orgs):\n",
    "    for (p2, path_org2) in enumerate(full_paths_orgs):\n",
    "        output_file = path_output_m + 'output_' + str(p1+1) + '_' + str(p2+1) + '.txt'\n",
    "        !fastANI -q {path_org1} -r {path_org2} -o {output_file} --fragLen {frag_len} -k {kmers} --minFrag {min_frag} --matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# result example in dir data/fastANI_data/orgs_gene_aligned_1066_fastANI_matrix"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Show results"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(144,\n",
       " ['fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_1.txt',\n",
       "  'fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_2.txt',\n",
       "  'fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_3.txt',\n",
       "  'fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_4.txt',\n",
       "  'fastANI_data/orgs_gene_aligned_1066_fastANI/output_1_5.txt'])"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "full_paths_output = [path_output + f for f in os.listdir(path_output) if f.endswith('.txt')]\n",
    "len(full_paths_output), full_paths_output[:5]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.concat([pd.read_csv(output_file, sep='\\t', header=None) for output_file in full_paths_output])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(144, 5)"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>99.9865</td>\n",
       "      <td>817</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_h...</td>\n",
       "      <td>83.3769</td>\n",
       "      <td>573</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_l...</td>\n",
       "      <td>88.3056</td>\n",
       "      <td>734</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_m...</td>\n",
       "      <td>83.1645</td>\n",
       "      <td>553</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_s...</td>\n",
       "      <td>fastANI_data/orgs_gene_aligned_1066/Bacillus_t...</td>\n",
       "      <td>83.2091</td>\n",
       "      <td>572</td>\n",
       "      <td>817</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   0  \\\n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...   \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...   \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...   \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...   \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...   \n",
       "\n",
       "                                                   1        2    3    4  \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_s...  99.9865  817  817  \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_h...  83.3769  573  817  \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_l...  88.3056  734  817  \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_m...  83.1645  553  817  \n",
       "0  fastANI_data/orgs_gene_aligned_1066/Bacillus_t...  83.2091  572  817  "
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "fastANI OUTPUT FORMAT: In all above use cases, OUTPUT_FILE will contain tab delimited row(s) with query genome, reference genome, ANI value, count of bidirectional fragment mappings, and total query fragments. Alignment fraction (wrt. the query genome) is simply the ratio of mappings and total fragments."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ani</th>\n",
       "      <th>count_bidir_frags</th>\n",
       "      <th>total_frags</th>\n",
       "      <th>organism_1</th>\n",
       "      <th>organism_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>99.9865</td>\n",
       "      <td>817</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>83.3769</td>\n",
       "      <td>573</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_halotolerans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>88.3056</td>\n",
       "      <td>734</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>83.1645</td>\n",
       "      <td>553</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_mojavensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>83.2091</td>\n",
       "      <td>572</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_tequilensis</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       ani  count_bidir_frags  total_frags           organism_1  \\\n",
       "0  99.9865                817          817  Bacillus_sonorensis   \n",
       "1  83.3769                573          817  Bacillus_sonorensis   \n",
       "2  88.3056                734          817  Bacillus_sonorensis   \n",
       "3  83.1645                553          817  Bacillus_sonorensis   \n",
       "4  83.2091                572          817  Bacillus_sonorensis   \n",
       "\n",
       "               organism_2  \n",
       "0     Bacillus_sonorensis  \n",
       "1   Bacillus_halotolerans  \n",
       "2  Bacillus_licheniformis  \n",
       "3     Bacillus_mojavensis  \n",
       "4    Bacillus_tequilensis  "
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['organism_1'] = df[0].apply(lambda x: '_'.join((x.split('/',2)[2].split('_',2)[:2])))\n",
    "df['organism_2'] = df[1].apply(lambda x: '_'.join((x.split('/',2)[2].split('_',2)[:2])))\n",
    "df.drop([0,1], axis=1, inplace=True)\n",
    "df.columns = ['ani', 'count_bidir_frags', 'total_frags', 'organism_1', 'organism_2']\n",
    "df.reset_index(inplace=True, drop=True)\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Most close organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ani</th>\n",
       "      <th>count_bidir_frags</th>\n",
       "      <th>total_frags</th>\n",
       "      <th>organism_1</th>\n",
       "      <th>organism_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>99.9930</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_halotolerans</td>\n",
       "      <td>Bacillus_halotolerans</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>130</th>\n",
       "      <td>99.9926</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_siamensis</td>\n",
       "      <td>Bacillus_siamensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>104</th>\n",
       "      <td>99.9925</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_velezensis</td>\n",
       "      <td>Bacillus_velezensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>99.9921</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_tequilensis</td>\n",
       "      <td>Bacillus_tequilensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>99.9909</td>\n",
       "      <td>819</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_paralicheniformis</td>\n",
       "      <td>Bacillus_paralicheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>99.9905</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "      <td>Bacillus_subtilis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>99.9888</td>\n",
       "      <td>819</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>99.9886</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_atrophaeus</td>\n",
       "      <td>Bacillus_atrophaeus</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>99.9865</td>\n",
       "      <td>817</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>99.9795</td>\n",
       "      <td>818</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_amyloliquefaciens</td>\n",
       "      <td>Bacillus_amyloliquefaciens</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ani  count_bidir_frags  total_frags                  organism_1  \\\n",
       "13   99.9930                818          820       Bacillus_halotolerans   \n",
       "130  99.9926                818          820          Bacillus_siamensis   \n",
       "104  99.9925                818          820         Bacillus_velezensis   \n",
       "52   99.9921                818          820        Bacillus_tequilensis   \n",
       "78   99.9909                819          820  Bacillus_paralicheniformis   \n",
       "91   99.9905                818          820           Bacillus_subtilis   \n",
       "26   99.9888                819          820      Bacillus_licheniformis   \n",
       "143  99.9886                818          820         Bacillus_atrophaeus   \n",
       "0    99.9865                817          817         Bacillus_sonorensis   \n",
       "65   99.9795                818          820  Bacillus_amyloliquefaciens   \n",
       "\n",
       "                     organism_2  \n",
       "13        Bacillus_halotolerans  \n",
       "130          Bacillus_siamensis  \n",
       "104         Bacillus_velezensis  \n",
       "52         Bacillus_tequilensis  \n",
       "78   Bacillus_paralicheniformis  \n",
       "91            Bacillus_subtilis  \n",
       "26       Bacillus_licheniformis  \n",
       "143         Bacillus_atrophaeus  \n",
       "0           Bacillus_sonorensis  \n",
       "65   Bacillus_amyloliquefaciens  "
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by=['ani'], ascending=False)[:10]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### Not very close organisms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ani</th>\n",
       "      <th>count_bidir_frags</th>\n",
       "      <th>total_frags</th>\n",
       "      <th>organism_1</th>\n",
       "      <th>organism_2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>80.8714</td>\n",
       "      <td>438</td>\n",
       "      <td>817</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>80.9150</td>\n",
       "      <td>426</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>110</th>\n",
       "      <td>81.0444</td>\n",
       "      <td>428</td>\n",
       "      <td>645</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>108</th>\n",
       "      <td>81.2449</td>\n",
       "      <td>418</td>\n",
       "      <td>645</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "      <td>Bacillus_sonorensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>81.2667</td>\n",
       "      <td>418</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_paralicheniformis</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>114</th>\n",
       "      <td>81.3444</td>\n",
       "      <td>415</td>\n",
       "      <td>645</td>\n",
       "      <td>Bacillus_vallismortis</td>\n",
       "      <td>Bacillus_paralicheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>82.9806</td>\n",
       "      <td>590</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_tequilensis</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>83.0327</td>\n",
       "      <td>585</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "      <td>Bacillus_tequilensis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>83.0471</td>\n",
       "      <td>552</td>\n",
       "      <td>805</td>\n",
       "      <td>Bacillus_mojavensis</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>83.0541</td>\n",
       "      <td>558</td>\n",
       "      <td>820</td>\n",
       "      <td>Bacillus_licheniformis</td>\n",
       "      <td>Bacillus_atrophaeus</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "         ani  count_bidir_frags  total_frags                  organism_1  \\\n",
       "9    80.8714                438          817         Bacillus_sonorensis   \n",
       "33   80.9150                426          820      Bacillus_licheniformis   \n",
       "110  81.0444                428          645       Bacillus_vallismortis   \n",
       "108  81.2449                418          645       Bacillus_vallismortis   \n",
       "81   81.2667                418          820  Bacillus_paralicheniformis   \n",
       "114  81.3444                415          645       Bacillus_vallismortis   \n",
       "50   82.9806                590          820        Bacillus_tequilensis   \n",
       "28   83.0327                585          820      Bacillus_licheniformis   \n",
       "38   83.0471                552          805         Bacillus_mojavensis   \n",
       "35   83.0541                558          820      Bacillus_licheniformis   \n",
       "\n",
       "                     organism_2  \n",
       "9         Bacillus_vallismortis  \n",
       "33        Bacillus_vallismortis  \n",
       "110      Bacillus_licheniformis  \n",
       "108         Bacillus_sonorensis  \n",
       "81        Bacillus_vallismortis  \n",
       "114  Bacillus_paralicheniformis  \n",
       "50       Bacillus_licheniformis  \n",
       "28         Bacillus_tequilensis  \n",
       "38       Bacillus_licheniformis  \n",
       "35          Bacillus_atrophaeus  "
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by=['ani'])[:10]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
